appcues/src/main/java/com/appcues/ui/AppcuesOverlayViewManager.kt

```diff
@@ -0,0 +1,93 @@
+package com.appcues.ui
+
+import android.app.Activity
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.ComposeView
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.findViewTreeLifecycleOwner
+import com.appcues.R
+import com.appcues.databinding.AppcuesOverlayLayoutBinding
+import com.appcues.logging.Logcues
+import com.appcues.monitor.AppcuesActivityMonitor
+import com.appcues.ui.composables.AppcuesComposition
+import com.appcues.ui.primitive.EmbedChromeClient
+import com.appcues.util.getNavigationBarHeight
+import com.appcues.util.getStatusBarHeight
+import org.koin.core.scope.Scope
+
+class AppcuesOverlayViewManager(private val scope: Scope) : DefaultLifecycleObserver {
```

Comment:
just like the DebuggerManager, it controls the lifecycle of the entire view in a way

Reply:
Did a first pass scan but wanted to step back and check my understanding on overall plan.

So this is moving toward inflating views directly over the customer view, inside of the customer app Activity, correct? Rather than the overlay Activity approach we had used with modals.

Are we thinking we'll be able to consolidate and share the approach across all pattern types moving forward? In particular, we've got the new layout .xml file now that is (i think) a copy of the old activity layout.xml. Then, we've got the current logic in the AppcuesActivity and more in the AppcuesViewModel - will we be able to share this VM in some fashion? AppcuesComposition now has the core of all the UI stuff in a nicely modularized spot, so assuming that would get shared.

Is it even technically possible to do tooltips in the overlay fashion like modals? or are we forced into the view inflating in the customer Activity to be able to effectively handle they element targeting coming up, keyholes, etc?

pardon all my questions, just trying to step back and understand the approach here as we lay out the new stuff - and partly jogging my memory from many days doing other stuff. We can have a little zoom on it maybe

Reply:
it is reasonable to question this at this point. I figured it would be safer to not change modals for now. until we have tooltips all working so we can make tests against one and the other to see if we lose anything by going the view inflation approach instead of the activity.

right now we are sharing the Composition and the view model. I created a new layout file just because of the ID basically. but it should be easy to make it all work later using only this approach

Reply:
ok, mostly just wanted to understand thought process and future plans - totally agree on doing things incrementally and maybe cycling back to modals after things are progressed here.

I have no issue with continuing to progress things here as you build out the new concepts. It sounds like we have a good handle on the core re-use potential with the major parts there, so I'm happy. The more we can make things implement content rendering "the same" - the better, IMO.